### ISSUE TITLE
Is batch normalization used anywhere?

### ISSUE THREAD

**Issue by apoorvumang** [uma-pi1/kge#131]
In implementations such as https://github.com/ibalazevic/TuckER batch normalization is used to speed up training. However this means that additional parameters (batch norm parameters) are needed at inference. Is this used in embedding training in kge? If so, how do I get the batch norm parameters?

**Comment by samuelbroscheit** [comment]
There are two things that are currently not provided by LookupEmbedder to fully recreate TuckER:

1. BatchNorm
2. Entity Bias (which was used by TuckER for WNRR; that is only documented in a TuckER github issue)

However, I made a search with the same setting as in our ICLR paper with RelationalTucker3 and I got similar numbers to TuckER without BatchNorm and Entity Bias. I cannot say something about the speed of convergence for datasets like FB15K-237 and WNRR with/without BatchNorm. The TuckER paper didn't do any comprehensive ablation so its effect is unclear.

**Comment by rgemulla** [comment]
Just for completeness: batch norm could also be put into the projection embedder (which may be more suitable then in the lookup embedder). PRs welcome!

**Comment by samuelbroscheit** [comment]
I was thinking about BatchNorm and Entity Bias in the LookupEmbedder because it would also go along if we would factor ConvE into seperate components (which is doing the same thing like TuckER).

**Comment by samuelbroscheit** [comment]
But we would need a proper plan what the best factorization would be ...

**Comment by apoorvumang** [comment]
Thanks
